### ISSUE TITLE
net-wireless/sdrtrunk-0.5.0_beta6 does not respect LDFLAGS

### ISSUE THREAD

**Issue by asarubbo** [pentoo/pentoo-overlay#1415]
https://blogs.gentoo.org/ago/2020/07/04/gentoo-tinderbox/

Issue: net-wireless/sdrtrunk-0.5.0_beta6 does not respect LDFLAGS.
Discovered on: amd64 (internal ref: pentoo_tinderbox)

NOTE:
(CLANG-STRICTER-SYSTEM) in the summary means that the bug was found on a machine that runs clang with stricter mode

This machine uses clang with CFLAGS/CXXFLAGS=-Werror=implicit-function-declaration -Werror=implicit-int -Werror=incompatible-function-pointer-types

See also:
https://archives.gentoo.org/gentoo-dev/message/dd9f2d3082b8b6f8dfbccb0639e6e240
https://wiki.gentoo.org/wiki/Modern_C_porting
To reproduce this issue you may want to set LDFLAGS="${LDFLAGS} -Wl,--defsym=__gentoo_check_ldflags__=0". If this is not something about c/c++ context you may want to see this bug as an hint to hide the QA warning (with QA_FLAGS_IGNORED) where is not possible to respect LDFLAGS.

At the time of writing, github-cli does not support attachments. I'm using gists but they can be deleted at a certain point. If you want to save them, please create attachments on your own.

build log: https://gist.github.com/086019a5fa34ad199ea8cc42acf90570

**Comment by asarubbo** [comment]
pentoo_tinderbox has reproduced this issue with version 0.5.0_beta6-r1 - Updating summary.

**Comment by asarubbo** [comment]
pentoo_tinderbox has reproduced this issue with version 0.5.2 - Updating summary.

**Comment by blshkv** [comment]
● The pattern opt/sdrtrunk/lib/*/* covers lib/client/libjvm.so but QA_PREBUILT doesn't suppress the flags check —
  need QA_FLAGS_IGNORED with a regex covering all levels:                                                           
                                                                                                                    
● Update(net-wireless/sdrtrunk-bin/sdrtrunk-bin-0.6.1-r1.ebuild)                                                    
  ⎿  Added 2 lines                                                                                                  
      47                                                                                                            
      48  QA_PREBUILT="opt/sdrtrunk/bin/* opt/sdrtrunk/lib/*/* opt/sdrtrunk/lib/*"                                  
      49                                                                                                            
      50 +QA_FLAGS_IGNORED="opt/sdrtrunk/.*"                                                                        
      51 +                                                                                                          
      52  QA_PRESTRIPPED="opt/sdrtrunk/lib/minimal/libjvm.so"                                                
      53   
      54  src_install() {

● Done. QA_FLAGS_IGNORED="opt/sdrtrunk/.*" is an extended regex that matches all pre-built files under
  /opt/sdrtrunk/ at any depth, suppressing the CFLAGS/LDFLAGS QA warning for bundled JVM binaries that obviously
  can't be built with Gentoo's flags.

**Comment by ZeroChaos-** [comment]
Why do we need the bundled stuff at all?  We are building this from source.

**Comment by blshkv** [comment]
yeah, that would be a proper solution. Ideally, the upstream should remove these java* related files. Patches are welcome as always.

**Comment by ZeroChaos-** [comment]
I thought you were fixing qa issues, this looks more like ignoring them and adding build failures...

**Comment by blshkv** [comment]
● Update(net-wireless/sdrtrunk/sdrtrunk-0.5.2-r2.ebuild)    
  ⎿  Added 1 line, removed 1 line
      57    GRADLE="gradle --gradle-user-home .gradle --console rich --no-daemon"                                   
      58    GRADLE="${GRADLE} --offline"                                                                     
      59    unset TERM                                                                                              
      60 -  ${GRADLE} runtime -x check -x test || die                                                        
      60 +  ${GRADLE} installDist -x check -x test || die                                                           
      61  }                                                                                                  
      62   
      63  src_install() {

● Let's verify the final result:                            

● Read 1 file (ctrl+o to expand)                                                                                    
                                                   
● Done. The two changes:                                                                                            
                                                            
  - gradle installDist — uses the application plugin to produce build/install/sdr-trunk/ with a shell launcher +    
  JARs, no bundled JVM                                                                                              
  - ./build/install/sdr-trunk/* — install path matches installDist output, no JVM binaries → no LDFLAGS/CFLAGS QA   
  warnings, QA_FLAGS_IGNORED not needed
